config/structs.go

```diff
@@ -1,12 +1,13 @@
 package config
 
 type ScheduledFeedConfig struct {
-	PubConfig    PublisherConfig `yaml:"publisher"`
-	EnabledFeeds []string        `yaml:"enabled_feeds"`
-	HttpPort     int             `yaml:"http_port,omitempty"`
+	PubConfig    TypeConfigPair   `yaml:"publisher"`
+	EnabledFeeds []string         `yaml:"enabled_feeds"`
+	Feeds        []TypeConfigPair `yaml:"feeds"`
```

Comment:
Struggling to visualise how the config file would look for this. As in, how would you know to list 'Feeds that require type/config' under `feeds`, and generic feeds under `enabled_feeds`? Maybe it's just a case of a using a different fieldname

Reply:
The idea here is perhaps preliminary but maintaining both EnabledFeeds and Feeds to allow configuration of these feeds (specifically pypi_critical in this case).

An example config file for this might look as follows:

```yaml
publisher:
  type: stdout

enabled_feeds:
  - pypi
  - rubygems
  - pypi_critical

feeds:
  - type: pypi_critical
    config:
      packages:
        - foopackage
        - barpackage
        - bazpackage
```

The whole type/config pair is something I'm keen to work around if there's a better option. The problem being the `packages` field containing a list of ScheduledFeed configurations but you can't unmarshal these interfaces without a hint towards the struct i.e. `type`